lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,267 @@
+      - Name of the plugin to operate on.
+    required: true
+    type: str
+    aliases:
+      - plugin_name
+
+  alias:
+    description:
+      - Alias of the plugin to operate on. Same plugin can be installed with different alias.
+    type: str
+    aliases:
+      - plugin_alias
+
+  plugin_options:
+    description:
+      - Dictionary of plugin settings.
+    type: dict
+
+  state:
+    description:
+      - C(absent) remove the plugin.
+      - C(present) install the plugin, if it does not already exist.
+    default: present
+    choices:
+      - absent
+      - present
+
+extends_documentation_fragment:
+  - docker
+  - docker.docker_py_2_documentation
+
+author:
+  - Vladimir Porshkevich (@porshkevich)
+
+requirements:
+  - "python >= 2.7"
+  - "docker-py >= 2.6.0"
+  - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+    module has been superseded by L(docker,https://pypi.org/project/docker/)
+    (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+    Version 2.1.0 or newer is only available with the C(docker) module."
+  - "Docker API >= 1.25"
+'''
+
+EXAMPLES = '''
+- name: Create a plugin
+  docker_plugin:
+    name: plugin_one
+
+- name: Remove a plugin
+  docker_plugin:
+    name: plugin_one
+    state: absent
+
+- name: Create a plugin with options
+  docker_plugin:
+    name: weaveworks/net-plugin:latest_release
+    alias: weave-net-plugin
+    plugin_options:
+      IPALLOC_RANGE: "10.32.0.0/12"
+      WEAVE_PASSWORD: XXXXXXXX
+'''
+
+RETURN = '''
+facts:
+    description: Plugin inspection results for the affected plugin.
+    returned: success
+    type: dict
+    sample: {}
+'''
+
+try:
+    from docker.errors import APIError, NotFound
+    from docker.models.plugins import Plugin
+    from docker import DockerClient
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+from ansible.module_utils.docker.common import DockerBaseClass, AnsibleDockerClient
+from ansible.module_utils.six import iteritems, text_type
+
+
+class TaskParameters(DockerBaseClass):
+    def __init__(self, client):
+        super(TaskParameters, self).__init__()
+        self.client = client
+
+        self.plugin_name = None
+        self.plugin_alias = None
+        self.plugin_options = None
+        self.debug = None
+
+        for key, value in iteritems(client.module.params):
+            setattr(self, key, value)
+
+
+class DockerPluginManager(object):
+
+    def __init__(self, client):
+        self.client = client
+
+        self.dclient = DockerClient()
+        self.dclient.api = client
+
+        self.parameters = TaskParameters(client)
+        self.check_mode = self.client.check_mode
+        self.results = {
+            u'changed': False,
+            u'actions': []
+        }
+        self.diff = self.client.module._diff
+
+        self.existing_plugin = self.get_existing_plugin()
+
+        state = self.parameters.state
```

Comment:
I would add at least 'enabled' to handle enable/disable state of the plugin

Reply:
Added